packages/edit-site/src/components/post-list/index.js

```diff
@@ -219,11 +220,27 @@ export default function PostList( { postType } ) {
 		},
 		[ location.path, location.query.isCustom, history ]
 	);
-
 	const getActiveViewFilters = ( views, match ) => {
 		const found = views.find( ( { slug } ) => slug === match );
 		return found?.filters ?? [];
 	};
+	useEffect( () => {
+		if ( isDirty ) {
+			return;
+		}
+
+		// The sort defaults are set in DEFAULT_POST_BASE.
+		if ( view.sort.field !== 'title' || view.sort.direction !== 'asc' ) {
+			setIsDirty( true );
+			setView( {
+				...view,
+				layout: {
+					...( view?.layout || {} ),
+					hierarchicalSort: false,
+				},
+			} );
+		}
+	}, [ isDirty, view ] );
```

Comment:
Why is this an effect instead of being handled on the `onChangeView` instead?

Reply:
🤔 The table layout is able to display the item level in any scenario, it doesn't need to disable it after the user sorts like this does.

The current approach makes this a consumer decision, enabling any other consumer to make it work differently. Giving it more thought, it seems more beneficial if the table layout absorbs this behaviour and it's opinionated about how it works with hierarchy levels.

Reply:
I was only commenting on the `useEffect` usage which we should avoid. You can still keep this at the user level but add the logic to the onChange callback no?

Reply:
Came here because of the `useEffect` too :)